rotkehlchen/db/dbhandler.py

```diff
@@ -2233,6 +2233,105 @@ def get_netvalue_data(
+
+        if from_ts is None:
+            from_ts = Timestamp(0)
+        if to_ts is None:
+            to_ts = ts_now()
+
+        # else
+        cursor.execute(
+            'SELECT COUNT(*) FROM timed_balances WHERE timestamp BETWEEN ? AND ?',
+            (from_ts, to_ts),
+        )
+        num_all_balances = cursor.fetchone()[0]
+
+        if len(balances) == num_all_balances:
+            return []
+
+        # else
+        asset_timestamps = [b.time for b in balances]
+        cursor.execute(
+            'SELECT timestamp, category FROM timed_balances WHERE timestamp BETWEEN ? AND ? '
+            'ORDER BY timestamp ASC',
+            (from_ts, to_ts),
+        )
+        all_timestamps, all_categories = zip(*cursor)
+        # dicts maintain insertion order in python 3.7+
+        timestamps_have_asset_balance = dict.fromkeys(all_timestamps, False)
+        timestamps_with_asset_balance = dict.fromkeys(asset_timestamps, True)
+        timestamps_have_asset_balance.update(timestamps_with_asset_balance)
+        prev_has_asset_balance = timestamps_have_asset_balance[all_timestamps[0]]  # the value of the first timestamp  # noqa: E501
+        prev_timestamp = all_timestamps[0]
+        inferred_balances: list[SingleDBAssetBalance] = []
+        for idx, (timestamp, has_asset_balance) in enumerate(timestamps_have_asset_balance.items()):  # noqa: E501
+            if idx == len(timestamps_have_asset_balance) - 1 and has_asset_balance is False:
+                # If there is no balance for the last timestamp add a zero balance.
+
+                # get the last known category of the asset
+                if len(all_timestamps) == 2:
+                    # If we are here, the first of the two snapshots is the asset one.
+                    # inferred_balances is empty, so we get the asset's category from the list
+                    category = all_categories[0]
+                else:
+                    # else we can get the last known category from the inferred_balances
+                    # It is the category of the asset at the start of the zero balance period
+                    category = inferred_balances[-1].category
+
+                inferred_balances.append(
+                    SingleDBAssetBalance(
+                        time=timestamp,
+                        amount=ZERO,
+                        usd_value=ZERO,
+                        category=category,
+                    ),
+                )
+            elif has_asset_balance is False and prev_has_asset_balance is True:
+                # add the start of a zero balance period
+                inferred_balances.append(
+                    SingleDBAssetBalance(
+                        time=timestamp,
+                        amount=ZERO,
+                        usd_value=ZERO,
+                        # the category of the previous timed_balance of the asset
+                        category=BalanceType.deserialize_from_db(all_categories[idx - 1]),
```

Comment:
In this cases I believe lefteris prefers

```suggestion
                        category=BalanceType.deserialize_from_db(all_categories[idx - 1]),  # noqa: E501  # the category of the previous timed_balance of the asset
```